lib/spack/spack/directives.py

```diff
@@ -180,24 +180,43 @@ def _depends_on(pkg, spec, when=None):
         when = pkg.name
     when_spec = parse_anonymous_spec(when, pkg.name)
 
+    if deptypes is None:
+        # The default deptype is build and link because the common case is to
+        # build against a library which then turns into a runtime dependency
+        # due to the linker.
```

Comment:
Probably not, actually.  The package *itself* does not call `mpirun` at runtime.  The user calls that *before* the app runs.  So, MPI is just a build and link dependency of the application.  A run dependency would be something like `curl` for `git`.